optlib/kotlin.ctags

```diff
@@ -0,0 +1,84 @@
+#
+#  Copyright (c) 2020, Jan Dolinár
+#
+#  Author: Jan Dolinár <dolik.rce@gmail.com>
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301,
+# USA.
+#
+
+# Regex for Kotlin
+--langdef=Kotlin
+--map-Kotlin=+.kt
+--map-Kotlin=+.kts
+
+--kinddef-Kotlin=c,class,classes
+--kinddef-Kotlin=o,object,objects
+--kinddef-Kotlin=i,interface,interfaces
+--kinddef-Kotlin=T,typealias,typealiases
+--kinddef-Kotlin=m,method,methods
+--kinddef-Kotlin=C,constant,constants
+--kinddef-Kotlin=v,variable,variables
+
+--_tabledef-Kotlin=toplevel
+--_tabledef-Kotlin=comment
+--_tabledef-Kotlin=string
+--_tabledef-Kotlin=mlstring
+--_tabledef-Kotlin=vals
+--_tabledef-Kotlin=vars
+
+# ignore anything in comment
+--_mtable-regex-Kotlin=toplevel/\/\*//{tenter=comment}
+--_mtable-regex-Kotlin=toplevel/\/\/[^\n]*\n//
+--_mtable-regex-Kotlin=comment/\*\///{tleave}
+--_mtable-regex-Kotlin=comment/[^*]+//
+--_mtable-regex-Kotlin=comment/.//
+
+# ignore anything in multiline strings
+--_mtable-regex-Kotlin=toplevel/"""//{tenter=mlstring}
+--_mtable-regex-Kotlin=mlstring/"""//{tleave}
+--_mtable-regex-Kotlin=mlstring/[^\\"]+//
+--_mtable-regex-Kotlin=mlstring/.//
+
+# ignore anything in strings
+--_mtable-regex-Kotlin=toplevel/"//{tenter=string}
+--_mtable-regex-Kotlin=string/[^\\"]+//
+--_mtable-regex-Kotlin=string/\\.//
+--_mtable-regex-Kotlin=string/"//{tleave}
+--_mtable-regex-Kotlin=string/.//
+
+# ignore character literals
+--_mtable-regex-Kotlin=toplevel/'.{1,2}'//
+
+# handle destructuring declarations
+--_mtable-regex-Kotlin=toplevel/val\s+\(\s*//{tenter=vals}
+--_mtable-regex-Kotlin=vals/([[:alnum:]_]+)\s*(:\s*[[:alnum:]_]+)*\s*,*\s*/\1/C/
+--_mtable-regex-Kotlin=vals/)//{tleave}
```

Comment:
`)` means the end of a pattern group. It requires `(`.
I guess what you wanted to write is `\)`.

Reply:
You're right, fixed. Thank you.